File: client-angular/src/models/message.ts
export class SendedMessage {
    token = "";
    text = "";
    time = new Date();
    author_id = 0;
    chat_id = 0;

    constructor(text="", time=new Date(), authorId=0) {
        this.text = text;
        this.time = time;
        this.author_id = authorId;
    }

}

export class Message {
    text = "";
    time = new Date();
    author_id = 0;
    chat_id = 0;

    constructor(text="", time=new Date(), authorId=0) {
        this.text = text;
        this.time = time;
        this.author_id = authorId;
    }

}

File: server/src/api/dto/message.go
package dto

import "time"

type MessageInput struct {
	Token        string        `json:"token"`
	AuthorID     int           `json:"author_id"`
	Text         string        `json:"text"`
	Attachements []Attachement `json:"attachements"`
	ChatID       int           `json:"chat_id"`
}

type Message struct {
	AuthorID     int           `json:"author_id"`
	CreatedAt    time.Time     `json:"time"`
	Text         string        `json:"text"`
	Attachements []Attachement `json:"attachements"`
	ChatID       int           `json:"chat_id"`
}
